InnerEye/ML/deep_learning_config.py

```diff
@@ -309,18 +309,38 @@ class DatasetParams(param.Parameterized):
                        "AzureML. Use an empty string for all datasets where a randomly chosen mount/download point "
                        "should be used.")
 
+    def validate(self) -> None:
+        if not self.azure_dataset_id and self.local_dataset is None:
+            raise ValueError("Either of local_dataset or azure_dataset_id must be set.")
+
+        if not self.azure_dataset_id and len(self.extra_azure_dataset_ids) > 0:
+            raise ValueError("Please specify an azure dataset in azure_dataset_id "
+                             "before adding datasets to extra_azure_dataset_ids.")
```

Comment:
maybe we are being too strict here? if someone specifies both datasets in extra..., and nothing in azure_dataset, it does not change the behaviour at all? The only thing that should matter: is it consistent across dataset_ids and mount points?